drivers/soundwire/intel_init.c

```diff
@@ -269,9 +269,11 @@ static struct sdw_intel_ctx
 
 err:
 	sdw_intel_cleanup(ctx);
-link_err:
 	driver_unregister(&intel_sdw_driver.driver);
 register_err:
+	kfree(ctx->links);
+	ctx->links = NULL;
```

Comment:
ok for the driver_unregister.

but for the free here it's not good. We already do a free in sdw_intel_cleanup, so maybe we remove the kfree from sdw_intel_cleanup and we also don't assign NULL since we free ctx below.

Reply:
@plbossart I think we need to free ctx->links in sdw_intel_cleanup() since sdw_intel_cleanup() will be used when normal exit. Maybe we can test ctx->links before freeing it?